spec/classes/init_spec.rb

```diff
@@ -232,14 +231,14 @@
               }
             end
 
-            content = <<-END.gsub(%r{^\s+\|}, '')
-              |set mail-format \{
-              |    from: monit\@test.local
-              |    message: Monit \$ACTION \$SERVICE at \$DATE on \$HOST: \$DESCRIPTION
+            content = <<~END
+              |set mail-format {
+              |    from: monit@test.local
+              |    message: Monit $ACTION $SERVICE at $DATE on $HOST: $DESCRIPTION
               |    subject: spectesting
-              |\}
+              |}
             END
-            it { is_expected.to contain_file('monit_config').with_content(%r{#{Regexp.escape(content)}}) }
+            it { is_expected.to contain_file('monit_config').with_content(%r{#{content}}) }
```

Comment:
I think the escaping is still needed. `%r{}` still creates a regex. If it's not a regex and it's the complete config then you can use
```suggestion
            it { is_expected.to contain_file('monit_config').with_content(content) }
```

I didn't try this, but this may also work if it's just substring testing, but I didn't test this myself.
```suggestion
            it { is_expected.to contain_file('monit_config').with_content(include(content)) }
```